source/server/http/admin.cc

```diff
@@ -417,11 +430,21 @@ void AdminFilter::onComplete() {
   ENVOY_STREAM_LOG(debug, "request complete: path: {}", *callbacks_, path);
 
   Buffer::OwnedImpl response;
-  Http::Code code = parent_.runCallback(path, response);
-
-  Http::HeaderMapPtr headers{
-      new Http::HeaderMapImpl{{Http::Headers::get().Status, std::to_string(enumToInt(code))}}};
-  callbacks_->encodeHeaders(std::move(headers), response.length() == 0);
+  Http::HeaderMapPtr header_map{new Http::HeaderMapImpl};
+  Http::Code code = parent_.runCallback(path, *header_map, response);
+  const auto& headers = Http::Headers::get();
+  header_map->addReferenceKey(headers.Status, std::to_string(enumToInt(code)));
+  if (header_map->ContentType() == nullptr) {
+    // Default to text-plain if unset.
+    header_map->addReferenceKey(headers.ContentType, "text/plain; charset=UTF-8");
```

Comment:
Some endpoints return JSON, should this be application/json in those cases?

Reply:
Yes, done.  Any others I should know about?